src/niamoto/core/plugins/exporters/html_page_exporter.py

```
@@ -0,0 +1,1135 @@
+                # Define the group-specific output directory prefix based on group_by_key
+                group_output_path_prefix = group_by_key
+                group_output_dir = output_dir / group_output_path_prefix
+
+                # Clear group directory only if filter matches or no filter is set
+                try:
+                    if (
+                        group_filter == group_by_key
+                    ):  # Only clear if this specific group is targeted
+                        if group_output_dir.exists() and any(
+                            group_output_dir.iterdir()
+                        ):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)